rules/DeadCode/Rector/If_/RemoveUnusedNonEmptyArrayBeforeForeachRector.php

```diff
@@ -156,4 +161,35 @@ private function isUselessBooleanAnd(BooleanAnd $booleanAnd, Expr $foreachExpr):
 
         return $this->countManipulator->isCounterHigherThanOne($booleanAnd->right, $foreachExpr);
     }
+
+    private function refactorStmtsAware(StmtsAwareInterface $stmtsAware): ?StmtsAwareInterface
+    {
+        foreach ((array) $stmtsAware->stmts as $key => $stmt) {
+            if (! $stmt instanceof If_) {
+                continue;
+            }
+
+            $nextStmt = $stmtsAware->stmts[$key + 1] ?? null;
+            if (! $nextStmt instanceof Foreach_) {
+                continue;
+            }
+
+            // the foreach must be the last one
+            if (isset($stmtsAware->stmts[$key + 2])) {
+                continue;
+            }
+
+            if (! $this->uselessIfCondBeforeForeachDetector->isMatchingEmptyAndForeachedExpr(
+                $stmt,
+                $nextStmt->expr
+            )) {
+                continue;
+            }
+
+            unset($stmtsAware->stmts[$key]);
+            return $stmtsAware;
```

Comment:
multiple if -> foreach can happen, this should continue to key + 2 call method recursively, with `$jumpToKey` usage, see 

https://github.com/rectorphp/rector-src/blob/edf54c85b6aadaecf064b797d340953697e0ca74/rules/CodingStyle/Rector/Stmt/NewlineAfterStatementRector.php#L118-L128

Reply:
I'm not sure how this works. Could you update it with a fixture and fix?

Reply:
I will try 👍

Reply:
I added patch for multi if and fixtures with jump to key+2 recursively https://github.com/rectorphp/rector-src/pull/3611/commits/9375b38bb9742c106f7d462272895d223be045d7